components/mediation-commons/src/main/java/org/wso2/carbon/mediation/commons/rest/api/swagger/OpenAPISpecification.java

```diff
@@ -0,0 +1,251 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mediation.commons.rest.api.swagger;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.swagger.v3.core.util.Json;
+import io.swagger.v3.core.util.Yaml;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.media.Content;
+import io.swagger.v3.oas.models.media.MediaType;
+import io.swagger.v3.oas.models.media.ObjectSchema;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.media.StringSchema;
+import io.swagger.v3.oas.models.parameters.PathParameter;
+import io.swagger.v3.oas.models.parameters.QueryParameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import io.swagger.v3.oas.models.servers.Server;
+import org.apache.axis2.AxisFault;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.api.API;
+import org.apache.synapse.api.version.URLBasedVersionStrategy;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PARAMETERS;
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PARAMETER_IN;
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PARAMETER_IN_BODY;
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PARAMETER_IN_PATH;
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PARAMETER_IN_QUERY;
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PARAMETER_NAME;
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PROTOCOL_HTTP_AND_HTTPS;
+import static org.wso2.carbon.mediation.commons.rest.api.swagger.SwaggerConstants.PROTOCOL_HTTP_ONLY;
+
+/**
+ * This class will generate the OAS 3.0 definition of the given synapse API.
+ */
+public class OpenAPISpecification {
```

Comment:
The class name doesn't make sense. I would suggest `OpenAPIProcessor` or something similar.

Reply:
+1 fixed